### PR TITLE
Code review comments

### PR DESCRIPTION
**Pull request by FredKSchott**
## Changes

- Make changes based on code review with some members of the team
- All small changes, mainly around documentation, code organization, and one babel improvement
- I had more review comments on the actual plugin itself, but will save that for another time!

### CODE REVIEW

File: packages/astro/src/vite-plugin-jsx/index.ts [hunk 1]
```diff
@@ -36,6 +39,58 @@ function isSSR(options: undefined | boolean | { ssr: boolean }): boolean {
   return false;
 }
 
+function getEsbuildLoader(fileExt: string): string {
```
[the remark this answers is not in the document]
Comment (matthewp) [reply packages/astro/src/vite-plugin-jsx/index.ts:42]: I think regardless of where the function is, a clearer name here is an improvement.

**Comment by drwpow** [comment]
Coming back to this: I didn’t mean to block! I’m neutral on this PR, and it can be merged. I just wanted to flag that in an open source project, trying to enforce one individual’s personal coding style on a collaborative repo will always be moot without a) establishing sweeping consistency, and b) making those rules known (ideally via linting). Feel free to merge this!

My goal was trying to advocate for better collaborative code quality by making expectations clear and not having one personal coding style overtake an already-consistent codebase. I apologize for my tone sounding negative or pushing against code changes here.

**Comment by matthewp** [comment]
How about a compromise here. I think this PR is clearly illustrating a few things:

- We do not have a consistent coding style in Astro.
- Some people are passionate about certain coding styles such as where functions live within a file or how comments are used (and when they are required).
- Some coding style things are enforceable via lint rules and some are not. Some disagree on how important that is.

Given that this PR has a few things that improve readability that I *think* we agree on, such as a clearer name for one of the functions, I say that we:

1. Merge this PR as is. I think if a community member submitted a similar PR we would likely accept it.
1. We start a `styleguide.md` with the Core team. We can come to a consensus on all of the above disagreements there, and if necessary have a vote.
1. Any future PRs we can always refer to that document and changes will be non-controversial.

**Comment by jonathantneal** [comment]
Thank you @drwpow for identifying the need so clearly, and thank you for the summary and actionable, @matthewp. I am also in favor of the styleguide, too. I’ve brought the style guide up to the maintainers channel on Discord.

With 2 approvals, @FredKSchott, could you merge this as-is?

**Comment by FredKSchott** [comment]
Yes! Sorry, I didn't mean to leave this around either. will merge and then follow up on any discussion that I've missed

**Comment by FredKSchott** [comment]
Yes, I'm +1 on @matthewp and @jonathantneal's suggestions to resolve @drwpow's concerns. A style guide would be a great thing, and I generally agree that anything that we all agree on that can be linted, should be linted.

fwiw I agree that if this PR came in from a community member without context, I'd be neutral on merging myself. But since this came out of a team exercise around reviewing style, I think in that context a style-only PR makes sense.  

I think there's an action item on me to circle back to the expectations around doing style reviews, now that we've done a couple. Reviewing coding style written by others on the team can be a fun exercise but it should also be towards a larger goal than just random PRs. Prompting discussions that can lead to the creation of a style guide is very much in line with what I had in mind for these review sessions!

**Comment by FredKSchott** [comment]
Also in case not clear: the merging of this PR does not mean that the things included in them are now rules that must be followed or else. I expect that everything not currently covered by either lint rules or a style guide is currently "undefined" at best.

**Pull request merged**